### PR TITLE
Add citation to scan kernel template documentation

### PR DESCRIPTION
**Pull request by adamfidel**
This PR adds a citation for decoupled lookback.

### CODE REVIEW

File: documentation/library_guide/kernel_templates/single_pass_scan.rst [hunk 1]
```diff
@@ -7,7 +7,8 @@ inclusive_scan Function Template
 
 The ``inclusive_scan`` function computes the inclusive prefix sum using a given binary operation.
 The function implements a single-pass algorithm, where each input element is read exactly once from
-global memory and each output element is written to exactly once in global memory.
+global memory and each output element is written to exactly once in global memory. This function
+is an implementation of the Decoupled Look-back [#fnote1]_ scan algorithm.
```
Comment (danhoeflinger) [inline documentation/library_guide/kernel_templates/single_pass_scan.rst:11]:
```suggestion
is an implementation of the decoupled look-back [#fnote1]_ scan algorithm.
```

File: documentation/library_guide/kernel_templates/single_pass_scan.rst [hunk 2]
```diff
@@ -202,7 +203,8 @@ The initial configuration may be selected according to these high-level guidelin
    Avoid setting too large ``param.data_per_workitem`` and ``param.workgroup_size`` values.
    Make sure that :ref:`Memory requirements <scan-memory-requirements>` are satisfied.
 
-.. [#fnote1] The X\ :sup:`e`-core term is described in the `oneAPI GPU Optimization Guide
+.. [#fnote1] Merrill, D., Garland, M.: Single-pass parallel prefix scan with decoupled look-back. Technical report NVR-2016-002, NVIDIA (2016)
```
Comment (danhoeflinger) [inline documentation/library_guide/kernel_templates/single_pass_scan.rst:206]: Should the title be capitalized?  @dcbenito

File: documentation/library_guide/kernel_templates/single_pass_scan.rst (hunk 1 above)
Comment (adamfidel) [reply documentation/library_guide/kernel_templates/single_pass_scan.rst:11]: I was trying to be consistent with the radix sort documentation, which capitalizes the word Onesweep.
https://github.com/oneapi-src/oneDPL/blob/a9aabb2b94020634f8f9961471f8af0a2aeb60ea/documentation/library_guide/kernel_templates/esimd/radix_sort.rst?plain=1#L10

Comment (danhoeflinger) [reply documentation/library_guide/kernel_templates/single_pass_scan.rst:11]: Aha ok.  Fair enough.

**Review by danhoeflinger — DISMISSED** [review]
LGTM.  I think I was wrong about my capitalization comments.

**Review by dmitriy-sobolev — DISMISSED** [review]
LGTM

**Review by dcbenito — DISMISSED** [review]
I went to the paper, and it should be capitalized. Please cite it as: Merrill, D., Garland, M.: Single-pass Parallel Prefix Scan with Decoupled Look-back. Technical Report NVR-2016-002, NVIDIA (2016)

Once that is fixed, I approve this PR.

**Review by dmitriy-sobolev — APPROVED** [review]
Reapproving after capitalization.

**Pull request merged**